ideafast/middleware-services
# Inventory datetime fix

As @davidverweij outlined in #79 it is possible for a user of snipe-it to button-smash and make N checkout/checkin requests at the same time. This then causes issues with existing logic, resulting in incorrect datetimes in the history of devices.

As such, I have modified the code to fix this bug. While the fix is potentially simple (one-line), I instead refactored the method because it was performing too much logic implictely, which while more efficient impacted readability. Likewise, I have taken this as a chance to pull out `serialize` methods to relevant schema classes.

### To Test

1. Review Code
2. Review Tests from C.I
3. Download locally and run `poetry run consumer` and navigate to to the device that caused the bug [http://0.0.0.0:8000/inventory/device/history/BTF-ABC](http://0.0.0.0:8000/inventory/device/history/BTF-ABC]). The history should now be resolved.

Closes #79
